ocaml/stdlib/array.mli

```diff
@@ -33,7 +33,7 @@ open! Stdlib
 type 'a t = 'a array
 (** An alias for the type of arrays. *)
 
-external length : 'a array -> int = "%array_length"
```

Comment:
I'm not sure we should be making this change. I think it might break attempts to build opam, or even just things in our tree. We mostly only ever use `[@local_opt]` or similar in the stdlib. Probably we should add `[@contended_opt]` or similar and do the same -- although not in this PR.

Reply:
This change *was* in #2398 but removed due to similar concerns. The current PR will not contain this change after rebase.

Reply:
I think the options were:
- Not do anything, because our internal base/core will have the proper interface anyway.
- Define those changed functions in `otherlibs`.